apps/snitch_core/test/data/model/stock/stock_transfer_test.exs

```diff
@@ -0,0 +1,78 @@
+defmodule Snitch.Data.Model.StockTransferTest do
+  @moduledoc false
+
+  use ExUnit.Case, async: true
+  use Snitch.DataCase
+  import Snitch.Factory
+  alias Snitch.Data.Model.StockTransfer, as: StockTransferModel
+
+  setup do
+    [stock_location: insert(:stock_location)]
+  end
+
+  describe "create/4" do
+    test "Fails for INVALID number id" do
+      assert {:error, changeset} = StockTransferModel.create("", "", -1, -1)
+      refute changeset.valid?
+      assert %{number: ["can't be blank"]} = errors_on(changeset)
+    end
+
+    test "Fails with invalid destinaton location" do
+      assert {:error, changeset} = StockTransferModel.create("", "T2132", nil, nil)
+      refute changeset.valid?
+      assert %{destination_location_id: ["can't be blank"]} = errors_on(changeset)
+    end
+
+    test "Inserts with valid attributes", context do
+      %{stock_location: stock_location} = context
+
+      assert {:ok, stock_transfer} =
+               StockTransferModel.create("", "T1235450", nil, stock_location.id)
+
+      assert stock_transfer.destination_location_id == stock_location.id
+    end
+  end
+
+  describe "get/1" do
+    test "Fails with invalid id" do
+      stock_item = StockTransferModel.get(1)
+      assert nil == stock_item
+    end
+
+    test "gets with valid id", context do
```

Comment:
IMO we should not spend any time in testing `get`, `get_all`, `delete`, if all they do is to delegate the work to `QueryHelper`.
This slows down our test suite because the tests here hit the DB quite a lot (each `stock_transfer` insert is 3 DB inserts).
We can improve coverage later, ideally by mocking away `QueryHelper` for these tests and keeping tests for namesake, because we'll test `QueryHelper` separately.